src/components/navigation/z-menu-section/styles.css

```diff
@@ -43,6 +46,34 @@
   text-align: left;
 }
 
+::slotted([data-text]) {
+  display: inline-flex;
+  flex-direction: column;
+}
+
+/* Trick to prevent layout shifts on font-weight changes.
+The trick is to put an hidden copy of the text already set to bold,
+so the needed space is correctly calculated before any hover/active state changes the
+font-weight of the real text.
+The hidden text has an height of 0px so it can expand the width without changing the height
+when placed below the real text using `flex-direction: column`. */
+::slotted([data-text])::after {
+  overflow: hidden;
+  height: 0;
+  content: attr(data-text);
+  content: attr(data-text) / "";
```

Comment:
in the line 64 I'm using the **alternative text** (doc [here](https://developer.mozilla.org/en-US/docs/Web/CSS/content)) for screen readers, setting and empty string to avoid reading the text as it's a duplicate. Line 63 is a fallback for browsers that doesn't support the alternative text and it's necessary because (quoting the docs)
> if the browser does not support alternative text, neither the content or alternative text will be used